third_party/rules_pycross/pycross/private/tools/wheel_installer.py

```diff
@@ -97,6 +98,30 @@ def main(args: Any) -> None:
 
     setup_namespace_pkg_compatibility(lib_dir)
 
+    if args.patch:
+        if not args.patch_tool and not args.patch_tool_target:
+            raise ValueError("Specify one of 'patch_tool' or 'patch_tool_target'.")
+
+        patch_args = [
+            args.patch_tool or Path.cwd() / args.patch_tool_target
+        ] + args.patch_arg
+        patch_dir = args.patch_dir or "."
+        for patch in args.patch:
+            with patch.open("r") as stdin:
```

Comment:
I cannot comment on whether it's universal, but at least GNU patch (and presumably BSD patch) is pretty much hard-coded to expect the patch on standard input. It's actually kind of annoying to make `patch` accept the patch as an argument instead of via stdin.

Unless I'm missing something simple? At least it appears not to be as simple as specifying the patch as an argument.

Reply:
Thanks for explaining, LGTM, I am dealing with patching on a daily basis.